databooks/data_models/notebook.py

```diff
@@ -80,18 +95,31 @@ def clear_metadata(
                 "Exactly one of `cell_metadata_keep` or `cell_metadata_remove` must"
                 f" be passed, got {nargs} arguments."
             )
+
         if cell_metadata_keep is not None:
             cell_metadata_remove = tuple(
                 field for field, _ in self.metadata if field not in cell_metadata_keep
             )
         self.metadata.remove_fields(cell_metadata_remove)  # type: ignore
 
-        self.remove_fields(remove_fields, missing_ok=True)
+        cell_fields = self.__fields__  # especified in Class definition
+        if any(field in cell_remove_fields for field in cell_fields):
+            logger.debug(
+                "Ignoring removal of "
+                + str([f for f in cell_remove_fields if f in cell_fields])
+                + f" - removing fields yields invalid `{type(self).__name__}`."
+            )
+            cell_remove_fields = [f for f in cell_remove_fields if f not in cell_fields]
+
+        super(Cell, self).remove_fields(cell_remove_fields, missing_ok=True)
+
         if self.cell_type == "code":
-            if cell_outputs:
-                self.outputs: List[Dict[str, Any]] = []
-            if cell_execution_count:
-                self.execution_count = None
+            self.outputs: List[Dict[str, Any]] = (
+                [] if "outputs" not in dict(self) else self.outputs
+            )
+            self.execution_count: Optional[PositiveInt] = (
+                None if "execution_count" not in dict(self) else self.execution_count
+            )
```

Comment:
It might be a good idea to move these notebook fixes into the remove_fields function itself. Then it becomes a "correct" function again that can be called without consequences.

Also since this code is related to removing fields I think it fits there better

Reply:
Done!